Document filtering sector types (#305)
filecoin-project/lotus-docs: content/en/storage-providers/operate/custom-storage-layout.md

This location can be made of large capacity, albeit slower, spinning-disks.

### Filter sector types

You can filter for what sector types that are allowed in each storage path by adjusting the configuration file in: `<path-to-storage>/sectorstorage.json`.

```json
{
  "ID": "1626519a-5e05-493b-aa7a-0af71612010b",
  "Weight": 10,
  "CanSeal": false,
  "CanStore": true,
  "MaxStorage": 0,
  "Groups": [],
  "AllowTo": [],
  "AllowTypes": null,
  "DenyTypes": null
}
```

Valid values for `AllowTypes` and `DenyTypes` are:

```
"unsealed"
"sealed"
"cache"
"update"
"update-cache"
```

Any other value will generate a warning and be ignored.

### Seperate sealed and unsealed

A very basic setup where you want to seperate unsealed and sealed sectors could be achieved by:

- Add `"DenyTypes": ["unsealed"]` to long-term storage path(s) where you want to store the sealed sectors.
- Add `"AllowTypes": ["unsealed"]` to long-term storage path(s) where you want to store the unsealed sectors.

> This setting will still allow for `cache` and `update-cache` files to be put in the storage path that allows for `unsealed` sectors in it. If you want to completly deny all other sector types then `unsealed` in this path, you should add all other valid values to the `"DenyTypes"` for that storage path.

Reviewer: ```suggestion
Setting only `unsealed` for `AllowTypes` will still allow `cache` and `update-cache` files to be placed in this storage path. If you want to completely deny all other types of sectors in this path, you can add additional valid values to the `"DenyTypes"` field.
```

{{< alert icon="tip">}}
If there are existing files with disallowed types in a storage path, those files will remain readable for PoSt/Retrieval. So the worst that can happen in case of misconfiguration in the storage path is that sealing tasks will get stuck waiting for storage to become available.
{{< /alert >}}

## Listing storage locations

You can see all your storage locations with the `lotus-miner storage list` command:

```shell
lotus-miner storage list

Sealing:
[##########                             ] 1.521 TiB/6.93 TiB 21%
  Unsealed: 1; Sealed: 2; Caches: 2; Reserved: 0 B
  Weight: 10; Use: Seal 
  Local: /root/sealing
  URL: http://127.0.0.1:2345/remote

Storage-Unsealed:
[##############                         ] 23.61 TiB/96.64 TiB 24%
  Unsealed: 411; Sealed: 0; Caches: 0; Reserved: 0 B
  Weight: 10; Use: Store
  Allow Types: unsealed
  Local: /root/storage3
  URL: http://127.0.0.1:2345/remote

Storage-Sealed:
[##############                         ] 27.61 TiB/96.64 TiB 28%
  Unsealed: 0; Sealed: 471; Caches: 471; Reserved: 0 B
  Weight: 10; Use: Store
  Allow Types: sealed
  Local: /root/storage2
  URL: http://127.0.0.1:2345/remote
```

### Detach storage paths

> You can detach a storage path with the `lotus-miner storage detach /path/to/storage` command. Please note that if you are detaching a storage path forever you should make sure that there are no leftover sectors in the storage path. 

Reviewer: ```suggestion
You can detach a storage path with the `lotus-miner storage detach /path/to/storage` command. Please note that if you are permanently detaching a storage path you should make sure that there are no leftover sectors in the storage path. 
```

## Updating locations

> You can update a sectors location with the `lotus-miner storage redeclare` command if you want to move sector data from one storage path to another storage path. If moving sectors to a storage path on a seperate server its recommended to use a utility like rsync or similar that has checksum on both ends. If moving data to a seperate server, it is also good practice to keep the important sector files like sealed sectors and cache until you have passed the first windowPoSt with the new storage location.

Reviewer: ```suggestion
You can update sector locations with the `lotus-miner storage redeclare` command if you want to move sector data from one storage path to another storage path. If moving sectors to a storage path on a separate server it's recommended to use a utility like r sync or similar that has checksum on both ends. If moving data to a separate server, it is also good practice to keep the important sector files like sealed sectors and cache until you have passed the first windowPoSt with the new storage location.
```
